frame/staking/src/tests.rs

```diff
@@ -5906,3 +5892,124 @@ mod staking_interface {
 		})
 	}
 }
+
+mod ledger {
+	use super::*;
+
+	#[test]
+	fn paired_account_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(Staking::bond(
+				RuntimeOrigin::signed(10),
+				100,
+				RewardDestination::Controller
+			));
+
+			assert_eq!(<Bonded<Test>>::get(&10), Some(10));
+			assert_eq!(
+				StakingLedger::<Test>::paired_account(StakingAccount::Controller(10)),
+				Some(10)
+			);
+			assert_eq!(StakingLedger::<Test>::paired_account(StakingAccount::Stash(10)), Some(10));
+
+			assert_eq!(<Bonded<Test>>::get(&42), None);
+			assert_eq!(StakingLedger::<Test>::paired_account(StakingAccount::Controller(42)), None);
+			assert_eq!(StakingLedger::<Test>::paired_account(StakingAccount::Stash(42)), None);
+
+			// bond manually stash with different controller. This is deprecated but the migration
```

Comment:
Good, it is important we still consider all scenarios with existing unique stash and controller pairs.